src/bson/mod.rs

```diff
@@ -36,11 +50,10 @@ impl From<u64> for U64 {
     }
 }
 
-#[allow(clippy::from_over_into)]
-impl Into<u64> for U64 {
-    fn into(self) -> u64 {
+impl From<U64> for u64 {
```

Comment:
since when this is legal?

Reply:
Wasn't it always?

Reply:
yeah but we don't control the trait nor the type